src/compressor/Compressor.h

```diff
@@ -73,11 +73,47 @@ class Compressor {
   QatAccel qat_accel;
 #endif
 
-  static const char* get_comp_alg_name(int a);
-  static boost::optional<CompressionAlgorithm> get_comp_alg_type(const std::string &s);
+  static inline const char* get_comp_alg_name(int a) {
+    auto p = std::find_if(std::cbegin(compression_algorithms), std::cend(compression_algorithms),
+        [a](const auto& kv) { return kv.second == a; });
 
-  static const char *get_comp_mode_name(int m);
-  static boost::optional<CompressionMode> get_comp_mode_type(const std::string &s);
+    if (std::cend(compression_algorithms) == p)
+    return "???"; // It would be nice to revise this...
+
+    return p->first;
+  }
+
+  static inline boost::optional<CompressionAlgorithm> get_comp_alg_type(const std::string &s) {
```

Comment:
is string_view allowed here? If so - I'd suggest passing that instead of string (and using a better name...)

Reply:
ditto. please explain the rationale in the commit message why you move the definition here.

Reply:
also, if this is necessary, i'd suggest put this change into a separate commit.